src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php

```diff
@@ -353,38 +401,34 @@ public function ordersBetweenDatesGroupByDate(array $configuration = [])
         $groupBy = substr($groupBy, 0, -1);
         $groupBy = str_replace(' ', ', ', $groupBy);
 
-        $queryBuilder = $this->getQueryBuilderBetweenDatesGroupByDate(
-            $configuration['start'],
-            $configuration['end'],
-            $groupBy);
+        $queryBuilder = $this->getEntityManager()->getConnection()->createQueryBuilder();
 
         $queryBuilder
             ->select('DATE(o.completed_at) as date', 'COUNT(o.id) as "Number of orders"')
-        ;
-
-        return $queryBuilder
-            ->execute()
-            ->fetchAll();
-    }
-
-    protected function getQueryBuilderBetweenDatesGroupByDate(\DateTime $from, \DateTime $to, $groupBy = 'Date(date)')
-    {
-        $queryBuilder = $this->getEntityManager()->getConnection()->createQueryBuilder();
-
-        return $queryBuilder
             ->from($this->getClassMetadata($this->_entityName)->getTableName(), 'o')
             ->where($queryBuilder->expr()->gte('o.completed_at', ':from'))
             ->andWhere($queryBuilder->expr()->lte('o.completed_at', ':to'))
-            ->setParameter('from', $from->format('Y-m-d H:i:s'))
-            ->setParameter('to', $to->format('Y-m-d H:i:s'))
+            ->setParameter('from', $configuration['start']->format('Y-m-d H:i:s'))
+            ->setParameter('to', $configuration['end']->format('Y-m-d H:i:s'))
             ->groupBy($groupBy)
             ->orderBy($groupBy)
         ;
+
+        return $queryBuilder
+            ->execute()
+            ->fetchAll()
+        ;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function findExpired(\DateTime $expiresAt, $state = OrderInterface::STATE_PENDING)
     {
-        $queryBuilder = $this->getQueryBuilder();
+        $queryBuilder = $this->createQueryBuilder('o')
+            ->leftJoin('o.items', 'item')
```

Comment:
I'd wait and consider using EAGER in a separate PR. It is a bit of magic contrary to having full control over what is joined.